### ISSUE TITLE
Gradient not working as expected

### ISSUE THREAD

**Issue by laura-chacon** [brentvatne/react-native-linear-gradient#102]
I have this code using linear gradient copied from the code given in the examples section:

```
import LinearGradient from 'react-native-linear-gradient';

const styles = StyleSheet.create({
  parent: {
    flex: 1,
    backgroundColor: colors.MAIN_BACKGROUND_COLOR
  },
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  linearGradient: {
    paddingLeft: 15,
    paddingRight: 15,
    borderRadius: 5
  },
  buttonText: {
    fontSize: 18,
    fontFamily: 'Gill Sans',
    textAlign: 'center',
    margin: 10,
    color: '#ffffff',
    backgroundColor: 'transparent',
  },
});
```
```
render() {
  return (
      <View style={styles.parent}>
        <View style={styles.container}>
          <LinearGradient colors={['#4c669f', '#3b5998', '#192f6a']} style={styles.linearGradient}>
            <Text style={styles.buttonText}>
              Sign in with Facebook
            </Text>
          </LinearGradient>
        </View>
      </View>);
}
```

And this is the result that I wasn't expecting:

![gradient](https://cloud.githubusercontent.com/assets/395528/17181998/5401737e-5422-11e6-99fb-713d02186717.png)

Does someone know the problem with my code and why is it not applying in the proper way? 
Thank you

**Comment by stovmascript** [comment]
I'm seeing the same thing on RN v0.30.0

**Comment by stovmascript** [comment]
@laura-chacon Did you rebuild your project? I just rebuilt mine and it works. Also mentioned in #82.

**Comment by stovmascript** [comment]
@laura-chacon I also had to do step 3 of https://github.com/brentvatne/react-native-linear-gradient#android manually and rebuild for Android. I used rnpm to link the project, but it evidently doesn't link things in MainApplication.java file, but still does it in MainActivity.java.

**Comment by gijoehosaphat** [comment]
I'm also seeing this on Android (not using iOS). I am using react-native: 0.30.0. I did manually perform the installation steps and have made the changes to MainApplication (like our other 15+ modules)

I am testing the same example laura-chacon used.

**Comment by hj1980** [comment]
I had the same problem (red border) on iOS, restarting (killing off the packager and simulator) worked for me.

**Comment by laura-chacon** [comment]
@stovmascript how do I rebuild my React Native project? This is what I have done:
1. kill simulator
2. `rm -fr $TMPDIR/react-*`
3. in XCode, Product -> Clean
4. restart computer
5. in XCode, Product -> Build

@hj1980 I restarted the simulator and also the packager. For restarting the packager I killed the packager from terminal and typed `npm start`.

**Comment by stovmascript** [comment]
@laura-chacon By rebuilding I meant `react-native run-ios` or `react-native run-android`, but you're steps did the same I think. You've linked the package according to the [docs](https://github.com/brentvatne/react-native-linear-gradient#add-it-to-your-project) right? Seems to me like it still isn't linked if it's not working after rebuilding.

**Comment by btkramm** [comment]
Same thing here. I installed it using `rnpm link` and added to `MainApplication.java` manualy, but a get the same red square :/ I've tried rebuilding and restarting the packager but nothing.

**Comment by laura-chacon** [comment]
@stovmascript What would be the exact steps to properly configure `react-native-linear-gradient`? Would this be enough for it to work on iOS?
```
$ npm install react-native-linear-gradient --save
$ rnpm link react-native-linear-gradient
# in XCode, Product -> Clean
# in XCode, Product -> Build
```
Thanks!

**Comment by stovmascript** [comment]
@laura-chacon Yep, that should be ok for iOS. If it doesn't work with `rnpm link`, I would:

1. `git checkout ios/`
2. Link the library [manually](https://github.com/brentvatne/react-native-linear-gradient#manually)
3. XCode Clean, Build

**Comment by teodors** [comment]
I had the same problem (red border) on android using RN version 0.32.

I just had to do step 3 for android manual installation. This fixed it for me. It seems that `rnpm` is linking in `MainApplication.java` not in `MainActivity.java` .

**Comment by stovmascript** [comment]
@teodors It's the other way around:
- MainActivity.java for react-native < 0.29
- MainApplication.java for react-native >= 0.29

**Comment by teodors** [comment]
@stovmascript On the note of `rnpm`. I find that even official RN docs still use `rnpm` and not `react-native link`. I find it very confusing. Also, I can't find nothing about `react-native link` on official RN docs or how to use it. Although, that is off the topic.

**Comment by stovmascript** [comment]
@teodors Yeah, it's all WIP - they are rewriting the CLI, some info about that can be found in recent release notes, but that's about it. The current CLI doesn't have much docs to my knowledge, but you can atleast list help with `react-native --help`.

**Comment by stowball** [comment]
Weird. Using rpnm v1.9.0 and react-native 0.35, the references were incorrectly added to MainActivity.java, so had to do it manually

**Comment by fredriksvemark** [comment]
I noticed that this component is not working with the [Navigation component](https://github.com/wix/react-native-navigation) created by wix. 

This is on Android. For iOS however, it's working just fine.

**Comment by varunrayen** [comment]
@fredriksvemark  Could you explain how you added a linear gradient to a navbar?

**Comment by fredriksvemark** [comment]
@varunrayen Using the Navigation component somehow makes the Linear Gradient component fail to render views properly. The result is the same as the image pasted by the author. 
Adding a linear gradient to a navbar was not the case.

**Comment by chrisrobbins** [comment]
@fredriksvemark is there a fix/hack for this?

**Comment by fredriksvemark** [comment]
@chrisrobbins I'm sorry to say that I never managed to fix this, mainly because the project I was part of was concluded. However, this was almost one year ago. Perhaps you can find the solution/alternative somewhere now. Good luck!
